### PR TITLE
Cruft

### PR DESCRIPTION
**Pull request by bandophahita**
Using [cruft](https://cruft.github.io/cruft/) to help synchronize template portions of each project

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -183,53 +182,55 @@ tomli = {version = ">=2.0.1", python = "<3.11"}
...
 pre-commit = {version = "*", optional = true}
 pytest = {version = "*", optional = true}
 pytest-mock = {version = "*", optional = true}
-ruff = {version = ">=0.1.13", optional = true}
+ruff = {version = ">=0.2.0", optional = true}
+sphinx = {version = "*", optional = true}
+sphinx-rtd-theme = {version = "*", optional = true}
+tox = {version = "*", optional = true}
+
+autodoc-pydantic = {version = "*", optional = true}
```
Comment (bandophahita) [inline pyproject.toml:197]: I move this block of deps down for easier template override handling.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -7,6 +7,7 @@ repos:
       - id: black
         language_version: python3.12
   - repo: https://github.com/astral-sh/ruff-pre-commit
+    # Ruff version.
```
Comment (perrygoy) [inline .pre-commit-config.yaml:10]: Is this comment necessary?

File: README.md [hunk 3]
```diff
@@ -63,9 +64,9 @@ You want to contribute? Great! Here are the things you should do before submitti
 
 1. Fork the repo and git clone your fork.
 1. `dev` install the project package:
-   1. `pip install -e .[dev]` 
-   1. Optional (poetry users):
-      1. `poetry install --extras dev`
+    1. `pip install -e .[dev]`
+    1. Optional (poetry users):
+        1. `poetry install --extras dev`
```
Comment (perrygoy) [inline README.md:69]: We live in 4-space indentations all our lives and somehow missed this 3-space one for years. Can we really even call ourselves Pythonistas??

**Review by perrygoy — APPROVED** [review]
I only had one small question mark, but it is not blocking. I'm excited to give this tool a fair shake!

File: .pre-commit-config.yaml (hunk 2 above)
Comment (bandophahita) [reply .pre-commit-config.yaml:10]: you put it in there... I figured it was useful.

**Pull request merged**